psutil/__init__.py

```diff
@@ -358,6 +358,11 @@ def _init(self, pid, _ignore_nsp=False):
         self._exitcode = _SENTINEL
         # cache creation time for later use in is_running() method
         try:
+            try:
+                _psplatform.cext._check_pid_range(pid)
+            except OverflowError:
+                raise NoSuchProcess(pid, msg="process PID out of range")
+
```

Comment:
Should we fail fast on overflowed PIDs? The `NoSuchProcess` error can be caught and ignored by `_ignore_nsp=True` in below.

Reply:
> That should never happen though, because the PID will come from subprocess.Popen, so it's always supposed to be a "real" / legit PID.

That makes sense. I have moved this block as requested.

Reply:
Xie xie. Merging.